pennylane/qchem/observable_hf.py

```diff
@@ -116,6 +125,16 @@ def qubit_observable(o_ferm, cutoff=1.0e-12):
                 ops = ops + op[1]
                 coeffs = qml.math.concatenate([coeffs, qml.math.array(op[0]) * o_ferm[0][n]])
 
+    if active_new_opmath():
+        ps = qml.dot(coeffs, ops, pauli=True)
+        ps.simplify(tol=cutoff)
+
+        if len(ps) == 0:
+            return qml.s_prod(
+                0, qml.Identity(ops[0].wires[0])
+            )  # use any op and any wire to represent the null op
+        return ps.operation()
```

Comment:
We create a `PauliSentence` and then a `Sum` from that? It doesn't create an overhead? Makes me think that why we need the `Sum` at all? Why not just return the `PauliSentence`?